diode-server/cmd/auth_manage/main.go

```diff
@@ -0,0 +1,280 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+
+	"github.com/netboxlabs/diode/diode-server/auth"
+	"github.com/netboxlabs/diode/diode-server/authutil"
+)
+
+// SubCommand is an interface for a cli subcommand.
+type SubCommand interface {
+	Run() error
+}
+
+const (
+	defaultTimeout = 10 * time.Second
+)
+
+// cliLog logs a message to stderr.
+func cliLog(message string) {
+	fmt.Fprintln(os.Stderr, message)
+}
+
+func main() {
+	var cfg auth.Config
+	envconfig.MustProcess("", &cfg)
+
+	authClientManager := auth.NewHydraClientManager(
+		cfg.OAuth2.AdminServerURL,
+		slog.New(slog.NewTextHandler(os.Stderr, nil)),
+	)
```

Comment:
If we're going to abstract auth managers to enable support for other OAuth2 servers, I think it would be beneficial to add a list of supported auth manager types (hydra to start with), and instantiate the client based on provided command arg with default `hydra` (if not provided) as the only one supported currently. Potentially premature optimisation, but will happen sooner than later.

Reply:
I think it would probably be more of a configuration option vs a command line arg? It relies on consuming auth proxy's (env) configuration.

Reply:
That's correct, authmanager depends on diode-auth proxy configuration after all, but it's depends strictly on OAuth public/admin URLs only with standardized `/oauth2/token` and `/oauth2/introspect` endpoints so technically is auth provider agnostic, no indication of what is it. We could solve this by expanding `auth.OAuth2Config` with `Provider` with default `hydra`, and instantiate authClientManager in `diode-server/cmd/auth_manage/main.go` depending on list of supported oauth providers and clients, making it clean and expandable.

Reply:
Yep, makes sense.  May want to think about enumerating the standard endpoints since grouping is not guaranteed / hyrda is a bit anomolous afaict --  but needs a bit of thought overall with more than hydra in mind.

Reply:
Sounds good!